src/main/java/mezz/jei/ingredients/IngredientFilter.java

```diff
@@ -90,14 +91,17 @@ public IngredientFilter(
 			this.elementSearch.registerPrefix(prefixInfo);
 		}
 
+		WeakReference<IngredientFilter> weakThis = new WeakReference<>(this);
```

Comment:
we can name it `weakIngredientFilter` and `ingredientFilter` instead of `weakThis` and `this2`